controllers/counselors.js

```diff
@@ -1,117 +1,121 @@
+  })
+};
 
-const getCounselorsByCountry = (req, res) => {
-    const country = req.params.country;
-    let filterCounselors = counselors.filter(item => item.country.toLowerCase().includes(country.toLowerCase()))
-    return res.json(filterCounselors);
-}
-
-const getCounselorsById = (req, res) => {
-    const id = req.params.id;
-    let filterCounselorsId = counselors.filter(item => item.id == id)
-    if (filterCounselorsId.length === 0){
-        return res.status(404).send(`Counselor with ID ${id} not found`);
+const getCounselorByCategoryOrAll = (req, res) => {
+ const categoryName = Object.keys(req.query);
+ if (Object.keys(req.query).length === 0) {
+  Counselor.find({})
+   .then(data => {
+    return res.send(data);
+   })
+   .catch(err => {
+    return res.status(500).send('Some error ocurred while retrieving all Counselors data.');
+   });
+ } else {
+  Counselor.find(req.query)
+   .then(data => {
+    if (Object.keys(data).length !== 0) {
+     return res.send(data);
+    } else {
+     return res.status(404).send(`Doesn't exist any Counselor with this ${categoryName}: ${req.query[categoryName]}.`);
     }
-    return res.json(filterCounselorsId);
-}
+   })
+   .catch(err => {
+    return res.status(500).send(`Some error ocurred while retrieving Counselors by this category ${categoryName}.`);
+   });
+ }
+};
 
 const createCounselor = (req, res) => {
-    const {
-        first_name,
-        last_name,
-        email
-    } = req.query;
-    if (!first_name || !last_name || !email){
-        return res.status(400).send('Missing data to create Counselor');
-    } 
-    const newCounselor = {
-        id: counselors.length + 1,
-        first_name: first_name,
-        last_name: last_name,
-        email: email,
-        gender: req.query.gender,
-        address: req.query.address,
-        city: req.query.city,
-        country: req.query.country,
-        phone: req.query.phone,
-        availability: {
-            from: req.query.from,
-            to: req.query.to,
-            day: req.query.day,
-        }
-    }
-    counselors.push(newCounselor);
-    fs.writeFile('./data/counselors.json', JSON.stringify(counselors), (err) => {
-        if (err) {
-            return res.status(400).send('Couldn\'t create Counselor file');
-        } else {
-            return res.status(201).send('New Counselor succesfully created');
-        }
-    })
-}
+ const counselor = new Counselor({
+  first_name: req.body.first_name,
+  last_name: req.body.last_name,
+  email: req.body.email,
+  gender: req.body.gender,
+  address: req.body.address,
+  birthday: req.body.birthday,
+  city: req.body.city,
+  country: req.body.country,
+  phone: req.body.phone,
+  availability: {
+   from: req.body.from,
+   to: req.body.to,
+   day: req.body.day,
+  }
+ });
+ if (counselor.first_name !== undefined || counselor.last_name !== undefined || counselor.email !== undefined) {
```

Comment:
You can do it this way:
```suggestion
 if (counselor.first_name && counselor.last_name && counselor.email) {
```
It will be even better if you just check on the req.body for those fields and you prevent the server from creating an instance when you might not need it.